ts/test/ts_project_worker/fstree.test.js

```diff
@@ -19,147 +19,149 @@ function createFile(p) {
-fs.writeFileSync(path.join(root, 'not_a_symlink_but_null', 'deep', "input.js"), "");
-tree.add("not_a_symlink_but_null/deep/input.js");
-assert.deepStrictEqual(tree.readDirectory("not_a_symlink_but_null"), ["deep"])
-
-
-
-createFile("symlink/to/me/input.js")
-fs.symlinkSync(path.join(root, "symlink"), path.join(root, "symlinked"), "dir");
-
-tree.add("symlink/to/me/input.js");
-tree.add("symlinked");
-assert.ok(tree.directoryExists("symlinked/to/me"));
-assert.ok(!tree.fileExists("symlinked/to/me"));
-assert.ok(!tree.fileExists("symlinked"));
-assert.ok(tree.directoryExists("symlinked"));
-assert.deepStrictEqual(tree.getDirectories("symlinked"), ["to"])
-assert.deepStrictEqual(tree.readDirectory("symlinked"), ["to"])
-assert.deepStrictEqual(tree.readDirectory("symlinked/to/me"), ["input.js"])
-
-
-tree.remove("symlinked");
-assert.ok(!tree.directoryExists("symlinked"));
-assert.ok(!tree.directoryExists("symlinked/to"));
-
-
-createFile("symlinked/to/input.js")
-tree.add("symlinked/to/input.js")
-assert.ok(tree.directoryExists("symlinked"));
-assert.ok(tree.directoryExists("symlinked/to"));
-assert.ok(tree.fileExists("symlinked/to/input.js"));
-
-tree.remove("symlinked/to/input.js");
-assert.ok(!tree.directoryExists("symlinked"));
-assert.ok(!tree.directoryExists("symlinked/to"));
-
-
-fs.mkdirSync(path.join(root, "dir"));
-fs.mkdirSync(path.join(root, "sym"));
-fs.symlinkSync(path.join(root, "dir"), path.join(root, "sym", "s1"), "dir");
-fs.symlinkSync(path.join(root, "dir"), path.join(root, "sym", "s2"), "dir");
-
-createFile("dir/input.js");
-tree.add("dir/input.js");
-createFile("sym/to/input.js");
-tree.add("sym/to/input.js");
-tree.add("sym/s1");
-tree.add("sym/s2");
-assert.deepStrictEqual(tree.getDirectories("sym"), ["to", "s1", "s2"])
-
-// Dangling
-fs.mkdirSync(path.join(root, "dangle"));
-fs.symlinkSync(path.join(root, "dangle"), path.join(root, "may_dangle"), "dir");
-tree.add("may_dangle")
-assert.equal(tree.directoryExists("may_dangle"), false)
-assert.equal(tree.directoryExists("may_dangle/underneath"), false)
-assert.equal(tree.fileExists("may_dangle"), false)
-assert.equal(tree.fileExists("may_dangle/file.js"), false)
-assert.deepEqual(tree.readDirectory("may_dangle"), [])
-assert.deepEqual(tree.getDirectories("may_dangle"), [])
-assert.deepEqual(tree.readDirectory("may_dangle/underneath"), [])
-assert.deepEqual(tree.getDirectories("may_dangle/underneath"), [])
-
-// Remove
-createFile("correct_remove/to/input.js");
-tree.add("correct_remove/to/input.js");
-createFile("correct_remove/to/path/input.js");
-tree.add("correct_remove/to/path/input.js")
-tree.remove("correct_remove/to/path/input.js");
-assert.ok(tree.fileExists("correct_remove/to/input.js"));
-assert.ok(tree.directoryExists("correct_remove/to"));
-assert.ok(!tree.directoryExists("correct_remove/to/path"));
-assert.ok(!tree.fileExists("correct_remove/to/path/input.js"));
+// createFile("tree/subtree/input.js");
```

Comment:
remove or uncomment?

Reply:
I haven't gotten to this file yet. I'll be keeping these. you can skip this for now